crates/router/src/core/payments/routing.rs

```diff
@@ -1202,3 +1208,105 @@ pub fn make_dsl_input_for_surcharge(
     };
     Ok(backend_input)
 }
+
+/// success based dynamic routing
+#[cfg(all(feature = "v1", feature = "dynamic_routing"))]
+pub async fn perform_success_based_routing(
+    state: &SessionState,
+    routable_connectors: Vec<api_routing::RoutableConnectorChoice>,
+    business_profile: &domain::Profile,
+    dynamic_routing_algorithm: serde_json::Value,
+) -> RoutingResult<Vec<api_routing::RoutableConnectorChoice>> {
+    let success_based_dynamic_routing_algo_ref: api_routing::DynamicRoutingAlgorithmRef =
+        business_profile
+            .dynamic_routing_algorithm
+            .clone()
+            .map(|val| val.parse_value("DynamicRoutingAlgorithmRef"))
+            .transpose()
+            .change_context(errors::RoutingError::DeserializationError {
+                from: "dynamic_routing_algorithm".to_string(),
+                to: "DynamicRoutingAlgorithmRef".to_string(),
+            })
+            .attach_printable(
+                "unable to deserialize dynamic_routing_algorithm from business profile to DynamicRoutingAlgorithmRef",
+            )?
+            .unwrap_or_default();
+
+    let success_based_algo_ref = success_based_dynamic_routing_algo_ref
+        .success_based_algorithm
+        .ok_or(errors::RoutingError::GenericNotFoundError { field: "success_based_algorithm".to_string() })
+        .attach_printable(
+            "success_based_algorithm not found in dynamic_routing_algorithm from business_profile table",
+        )?;
+
+    if success_based_algo_ref.enabled_feature
+        == api_routing::SuccessBasedRoutingFeatures::DynamicConnectorSelection
+    {
+        let client = state
+            .grpc_client
+            .dynamic_routing
+            .success_rate_client
+            .as_ref()
+            .ok_or(errors::RoutingError::SuccessRateClientInitializationError)
+            .attach_printable("success_rate gRPC client not found")?;
+
+        let success_based_routing_configs = routing::helpers::fetch_success_based_routing_configs(
+            state,
+            business_profile,
+            dynamic_routing_algorithm,
```

Comment:
you are again passing the serialised `DynamicRoutingAlgorithmRef` to this function inside which `parse_value` is being called again. But you have already deserialised above. can we resolve this redundancy?

Reply:
So the function `fetch_success_based_routing_configs` is being called from one another function which isn't deserializing the `dynamic_routing_configs` (`push_metrics_for_success_based_routing`), that's why there is a redundant call. We can deserialize in another function separately as well to avoid the redundancy, will that be suffice?